webview/platforms/winui3.py

```diff
@@ -0,0 +1,1337 @@
+        settings.is_swipe_navigation_enabled = False
+        settings.is_zoom_control_enabled = True
+
+        if _state["user_agent"]:
+            settings.user_agent = _state["user_agent"]
+
+        if _state["private_mode"]:
+            # cookies persist even if UserDataFolder is in memory. We have to delete cookies manually.
+            sender.core_webview2.cookie_manager.delete_all_cookies()
+
+        if self.pywebview_window.real_url:
+            self.load_url(self.pywebview_window.real_url)
+        elif self.pywebview_window.html:
+            self.html = self.pywebview_window.html
+            self.load_html(self.pywebview_window.html, "")
+        else:
+            self.load_html(DEFAULT_HTML, "")
+
+        if _state["debug"] and webview_settings["OPEN_DEVTOOLS_IN_DEBUG"]:
+            sender.core_webview2.open_dev_tools_window()
+
+    def on_download_starting(
+        self, sender: CoreWebView2, args: CoreWebView2DownloadStartingEventArgs
+    ):
+        args.cancel = True
+
+        if not webview_settings["ALLOW_DOWNLOADS"]:
+            return
+
+        path = show_save_file_dialog(
+            self.form.app_window.id.value,
+            os.path.basename(args.result_file_path),
+            self.pywebview_window.localization["windows.fileFilter.allFiles"],
+            "*.*",
+        )
+        if path is None:
+            return
+
+        args.result_file_path = path
+        args.cancel = False
+
+    def on_navigation_start(
+        self, sender: WebView2, args: CoreWebView2NavigationStartingEventArgs
+    ):
+        pass
+
+    def on_navigation_completed(
+        self, sender: WebView2, _: CoreWebView2NavigationCompletedEventArgs
+    ):
+        url = str(sender.source)
+        self.url = None if self.ishtml else url
+
+        inject_pywebview(renderer, self.pywebview_window)
+
+
+_WINDOW_XAML = """
+<Grid xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation">
+    <Grid.RowDefinitions>
+        <RowDefinition Height="Auto"/>
+        <RowDefinition Height="*"/>
+    </Grid.RowDefinitions>
+    <Grid.ColumnDefinitions>
+        <ColumnDefinition Width="*"/>
+        <ColumnDefinition Width="Auto"/>
+    </Grid.ColumnDefinitions>
+
+    <MenuBar Name="menu" Grid.Row="0" Grid.ColumnSpan="2" Visibility="collapsed"/>
+    <WebView2 Name="webview" Grid.Row="1" Grid.ColumnSpan="2"
+        HorizontalAlignment="Stretch" VerticalAlignment="Stretch"/>
+</Grid>
+"""
+
+
+class BrowserView:
+    instances: dict[str, "BrowserForm"] = {}
+
+    app_menu_list: Iterable[Menu] = []
+
+    class BrowserForm:
+        def __init__(self, window: _Window, cache_dir: str):
+            self._is_active = False
+            self.uid = window.uid
+            self.pywebview_window = window
+            self.window = Window()
+            self.handle = get_window_from_window_id(self.window.app_window.id)
+            self.real_url = None
+
+            self.pywebview_window.native = self

```

Comment:
native should point to a native window (form) instance, ie to `self.window`
Additionally  self.pywebview_window.native.webview should point to a webview object

Reply:
fixed